src/test/java/com/revature/controllers/CohortControllerTest.java

```diff
@@ -1,30 +1,170 @@
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import javax.annotation.Resource;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+
+import com.revature.models.Address;
+import com.revature.models.Cohort;
+import com.revature.models.User;
+import com.revature.repos.CohortRepo;
+import com.revature.repos.UserRepo;
+import com.revature.services.CohortService;
+
+@RunWith(MockitoJUnitRunner.class)
 public class CohortControllerTest {
 
-//	@Test
-//	public void testFindByTrainer() {
-//		fail("Not yet implemented");
-//	}
-//
-//	@Test
-//	public void testFindAll() {
-//		fail("Not yet implemented");
-//	}
-//
-//	@Test
-//	public void testSave() {
-//		fail("Not yet implemented");
-//	}
-//
-//	@Test
-//	public void testJoinCohort() {
-//		fail("Not yet implemented");
-//	}
-//
-//	@Test
-//	public void testHandleError() {
-//		fail("Not yet implemented");
-//	}
+	@InjectMocks
+	CohortController cohortControllerTester;
+	
+	@Mock
+	CohortService cohortService;
+	
+	@Mock
+	CohortRepo cohortRepo;
+	
+	@Mock
+	UserRepo userRepo;
+	
+	@Mock
+	Page<Cohort> cohortsPage;
+	
+	@Test
+	public void testFindByTrainer() {
+//		Create a cohort stub
+		Cohort cohort = new Cohort();
+//		Create a list with that cohort on it
+		List<Cohort> cohorts = new ArrayList<>();
+		cohorts.add(cohort);
+//		Create fake id
+		int fakeTrainerId = (int) (Math.random() * 100);
+//		Define what the service method should return when passing the trainer id
+		when(cohortService.findByTrainer(fakeTrainerId)).thenReturn(cohorts);
+//		Create assertion
+		assertThat(cohortControllerTester.findByTrainer(fakeTrainerId)).isEqualTo(cohorts);
+	}
+
+	@Test
+	public void testFindAllByPage() {
+//		Create stub for a list of cohorts
+		List<Cohort> cohortsList = new ArrayList<>();
+//		Create a stub for the pageable parameter on the cohortsService.findAllByPage() method
+		Pageable pageable = PageRequest.of(1, 7, Sort.by("cohortId"));
+//		Define what each method should return
+		when(cohortsPage.getContent()).thenReturn(cohortsList);
+		when(cohortService.findAllByPage(pageable)).thenReturn(cohortsPage);
+//		Call the findAll method from the cohortController mock
+		Page<Cohort> result = cohortControllerTester.findAll(1);
+		List<Cohort> resultCohort = result.getContent();
+//		Write assertion
+		assertThat(cohortsList).isEqualTo(resultCohort);
+	}
+
+	@Test
+	public void testFindCohortUsers() {
```

Comment:
I don't see any stubs for `cohortService.findCohortUsers` here - how is the test being isolated?